src/Npgsql/TypeHandlerRegistry.cs

```diff
@@ -595,18 +595,14 @@ internal bool TryGetByOID(uint oid, out TypeHandler handler)
                 if (ByType.TryGetValue(type, out handler))
                     return handler;
 
-                Type arrayElementType = null;
+                PostgresType postgresType;
```

Comment:
Yep. It's a welcome change regardless, regular non-array types are much more frequent, it makes sense to handle them first.